common/buildcraft/core/marker/volume/VolumeBox.java

```diff
@@ -66,13 +71,15 @@ public VolumeBox(NBTTagCompound nbt) {
         }
         NBTTagList addonsTag = nbt.getTagList("addons", Constants.NBT.TAG_COMPOUND);
         IntStream.range(0, addonsTag.tagCount()).mapToObj(addonsTag::getCompoundTagAt).forEach(addonsEntryTag -> {
-            Class<? extends Addon> addonClass = AddonsRegistry.INSTANCE.getClassByName(new ResourceLocation(addonsEntryTag.getString("addonClass")));
+            Class<? extends Addon> addonClass =
+                AddonsRegistry.INSTANCE.getClassByName(new ResourceLocation(addonsEntryTag.getString("addonClass")));
             try {
                 Addon addon = addonClass.newInstance();
                 addon.box = this;
                 addon.readFromNBT(addonsEntryTag.getCompoundTag("addonData"));
                 EnumAddonSlot slot = NBTUtilBC.readEnum(addonsEntryTag.getTag("slot"), EnumAddonSlot.class);
                 addons.put(slot, addon);
+                addon.onAdded();
```

Comment:
Probably wrong, this method should be called when player actually adds addon, not when deserializing it from NBT.

Reply:
I've fixed this by adding a new method "postReadFromNbt"